dash/include/dash/Team.h

```diff
@@ -611,7 +672,8 @@ class Team
   mutable team_unit_t     _myid         = UNDEFINED_TEAM_UNIT_ID;
   mutable size_t          _size         = 0;
   Team                  * _parent       = nullptr;
-  Team                  * _child        = nullptr;
+  std::vector<Team*>      _children;
+  std::mutex              _mutex;
```

Comment:
`mutable`?

Reply:
It should be mutable once we have `const` Teams :D

Reply:
I only referenced the `_mutex` (see the _M&M rule_ [here](https://en.cppreference.com/w/cpp/language/cv)). Also, I think that this **and** the `_children` would be a prerequisite for having `const` Teams in the future.